### PR TITLE
Adding new properties to Data.Nat.Order.Properties

### PR DESCRIPTION
**Pull request by Alex1005a**
Adding three new properties to Data.Nat.Order.Properties and carried out a small refactoring in Data.Nat.Division.

### CODE REVIEW

File: libs/contrib/Data/Nat/Order/Properties.idr [hunk 1]
```diff
@@ -105,3 +105,23 @@ multLteMonotoneLeft a b r a_lt_b =
   rewrite multCommutative a r in
   rewrite multCommutative b r in
   multLteMonotoneRight r a b a_lt_b
+
+-- If a number is both GTE and LTE to another number, then they are equal
+export
+oppositeLteEq : LTE a b -> GTE a b -> a = b
```
Comment (Z-snails) [inline libs/contrib/Data/Nat/Order/Properties.idr:111]: This is the same as `antisymmetric` (https://github.com/idris-lang/Idris2/blob/main/libs/base/Data/Nat.idr#L114)

File: libs/contrib/Data/Nat/Order/Properties.idr [hunk 2]
```diff
@@ -105,3 +105,23 @@ multLteMonotoneLeft a b r a_lt_b =
...
+
+export
+lteNotLtEq : (a, b : Nat) -> LTE a b -> Not (LT a b) -> a = b
+lteNotLtEq a b a_lte_b not_n_lte_n = 
+    let b_lte_a = notLTImpliesGTE not_n_lte_n 
+    in oppositeLteEq a_lte_b b_lte_a
+
+-- Try succ left element LTE. Returns LT if successful, otherwise proof of equality a and b
+export
+succLeftLte : (a, b : Nat) -> LTE a b -> Either (LT a b) (a = b)
```
Comment (Z-snails) [inline libs/contrib/Data/Nat/Order/Properties.idr:124]: In my opinion this isn't the most intuitive name, maybe something like `decomposeLte` or `lteIsLtOrEq` (I prefer the former)

**Comment by Z-snails** [comment]
Could you fix the style issues the linter has found please? There's some trailing spaces at the end of lines, and there's not a newline at the end of the file.
If I remember correctly, you can run the linter locally with `python3 lint/lint.py`

**Comment by Alex1005a** [comment]
> Could you fix the style issues the linter has found please? There's some trailing spaces at the end of lines, and there's not a newline at the end of the file.

Sorry, forgot to fix this in the past commit.

**Pull request merged**